openfl-workspace/torch_cnn_mnist/src/dataloader.py

```diff
@@ -46,93 +52,58 @@ def __init__(self, data_path, batch_size, **kwargs):
         self.y_valid = y_valid
         self.val_loader = self.get_valid_loader()
 
-        self.num_classes = num_classes
-
-
-def load_mnist_shard(
-    shard_num, collaborator_count, categorical=False, channels_last=True, **kwargs
-):
+def load_dataset(data_path, train_split_ratio=0.8, **kwargs):
     """
-    Load the MNIST dataset.
-
+    Load your dataset here.
+    This function should be implemented to load the dataset from the given `data_path`.
+    You can use additional arguments passed via `**kwargs` if necessary.
     Args:
-        shard_num (int): The shard to use from the dataset
-        collaborator_count (int): The number of collaborators in the
-                                  federation
-        categorical (bool): True = convert the labels to one-hot encoded
-                            vectors (Default = True)
-        channels_last (bool): True = The input images have the channels
-                              last (Default = True)
-        **kwargs: Additional parameters to pass to the function
-
+        data_path (str): Path to the data directory.
+        **kwargs: Additional arguments that may be defined in `plan.yaml`
     Returns:
-        list: The input shape
-        int: The number of classes
-        numpy.ndarray: The training data
-        numpy.ndarray: The training labels
-        numpy.ndarray: The validation data
-        numpy.ndarray: The validation labels
+        Tuple containing:
+        - numpy.ndarray: The training data.
+        - numpy.ndarray: The training labels.
+        - numpy.ndarray: The validation data.
+        - numpy.ndarray: The validation labels.
     """
-    num_classes = 10
-
-    (X_train, y_train), (X_valid, y_valid) = _load_raw_datashards(
-        shard_num, collaborator_count, transform=transforms.ToTensor()
+    # Implement dataset loading logic here and return the appropriate data.
+    # Replace the following placeholders with actual data loading code.
+    dataset = MNISTDataset(
+        root=data_path,
+        transform=Compose([Grayscale(num_output_channels=1), ToTensor()])
     )
```

Comment:
This change would break the [quick-start tutorial](https://openfl.readthedocs.io/en/latest/get_started/quickstart.html#quick-start), which originally assumes that `torch_cnn_mnist` works "out of the box" (due to its reliance on an in-memory data loader). One option would be to update the quick-start tutorial to "install" the data set from openfl-contrib. Another approach would be to change the quick-start tutorial to use `torch_cnn_histology` which is also in-memory AFAIK.